accepted/2021/default_string_globalization.md

```diff
@@ -0,0 +1,390 @@
+
+### List of API obsoletions and additions
+
+In the diff below, a `-` indicates hiding; and a `+` indicates addition. A subset of `-` APIs are obsoleted, using the pattern described above.
+
+```diff
+ public sealed partial class String : System.Collections.Generic.IEnumerable<char>, System.Collections.IEnumerable, System.ICloneable, System.IComparable, System.IComparable<string?>, System.IConvertible, System.IEquatable<string?>
+ {
+     public static readonly string Empty;
+     [System.CLSCompliantAttribute(false)]
+     public unsafe String(char* value) { }
+     [System.CLSCompliantAttribute(false)]
+     public unsafe String(char* value, int startIndex, int length) { }
+     public String(char c, int count) { }
+     public String(char[]? value) { }
+     public String(char[] value, int startIndex, int length) { }
+     public String(System.ReadOnlySpan<char> value) { }
+     [System.CLSCompliantAttribute(false)]
+     public unsafe String(sbyte* value) { }
+     [System.CLSCompliantAttribute(false)]
+     public unsafe String(sbyte* value, int startIndex, int length) { }
+     [System.CLSCompliantAttribute(false)]
+     public unsafe String(sbyte* value, int startIndex, int length, System.Text.Encoding enc) { }
+     [System.Runtime.CompilerServices.IndexerName("Chars")]
+     public char this[int index] { get { throw null; } }
+     public int Length { get { throw null; } }
+     public object Clone() { throw null; }
+-    public static int Compare(System.String? strA, int indexA, System.String? strB, int indexB, int length) { throw null; }
+-    public static int Compare(System.String? strA, int indexA, System.String? strB, int indexB, int length, bool ignoreCase) { throw null; }
+-    public static int Compare(System.String? strA, int indexA, System.String? strB, int indexB, int length, bool ignoreCase, System.Globalization.CultureInfo? culture) { throw null; }
+-    public static int Compare(System.String? strA, int indexA, System.String? strB, int indexB, int length, System.Globalization.CultureInfo? culture, System.Globalization.CompareOptions options) { throw null; }
+     public static int Compare(System.String? strA, int indexA, System.String? strB, int indexB, int length, System.StringComparison comparisonType) { throw null; }
+-    public static int Compare(System.String? strA, System.String? strB) { throw null; }
+-    public static int Compare(System.String? strA, System.String? strB, bool ignoreCase) { throw null; }
+-    public static int Compare(System.String? strA, System.String? strB, bool ignoreCase, System.Globalization.CultureInfo? culture) { throw null; }
+-    public static int Compare(System.String? strA, System.String? strB, System.Globalization.CultureInfo? culture, System.Globalization.CompareOptions options) { throw null; }
++    public static int CompareString(System.String? strA, System.String? strB) { throw null; }
```

Comment:
It feels weird that `string` will be two times in the sentence when calling these APIs (`string.CompareString`) I guess there is no better naming/options?... I definitely like this better than including `Ordinal` in the name though.

Reply:
Downside is that the good names were taken by the method we want to steer people away from. :(

Reply:
Yeah this is the thing that urks me. Our goal is to present the developer with a simpler view of the world, but the new names look fairly misplaced and odd :-(

Reply:
In PowerShell repo we follow simple rule - "use explicitly culture/ordinal/ignorecase everywhere".
You will find that this code is full of explicit Ordinal and OrdinalIgnorecase names
This makes the code incredibly understandable and maintainable.
Despite this, we've fixed a few case of unexpected use of culture. So the PR with new conception is great!

But the new names looks weird for me too.
As I said, having Ordinal and OrdinalIgnorecase names in PowerShell code makes it better in my opinion.
Notice, we already have CompareOrdinal(). Why not follow this pattern?
```
CompareToString -> CompareToOrdinal
CompareToStringIgnoreCase -> CompareToOrdinalIgnoreCase
```
And this is completely consistent with the new concept. 
It also makes it easier to learn and understand for people coming from other languages.